v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -252,7 +253,12 @@ func writeToBagItem(
 	_, sourceIsMap := astmodel.AsMapType(actualSourceType)
 	_, sourceIsSlice := astmodel.AsArrayType(actualSourceType)
 
-	return func(reader dst.Expr, _ func(dst.Expr) []dst.Stmt, knownLocals *astmodel.KnownLocalsSet, generationContext *astmodel.CodeGenerationContext) []dst.Stmt {
+	return func(
+		reader dst.Expr,
+		_ func(dst.Expr) []dst.Stmt,
+		knownLocals *astmodel.KnownLocalsSet,
+		generationContext *astmodel.CodeGenerationContext,
+	) ([]dst.Stmt, error) {
```

Comment:
Not sure if its right to return nil as an error entirely throughout the function, but seems like we have to do it to make the function in line with the function type.

Reply:
Yeah, it's a common wart, caused when only some implementations of an interface need to return errors.